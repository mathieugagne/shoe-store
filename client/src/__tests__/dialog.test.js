import React from 'react';
import { shallow } from 'enzyme';
import toJson from 'enzyme-to-json';
import Dialog from '../dialog';

describe('dialog.js', () => {
  it('renders', () => {
    const handleClose = jest.fn();
    const wrapper = shallow(<Dialog
      title="Test Title"
      data={{}}
      handleClose={handleClose}
    />);
    expect(toJson(wrapper)).toMatchSnapshot();
    wrapper.find('WithStyles(Button)').simulate('click');
    expect(handleClose).toHaveBeenCalled();
  });
});
