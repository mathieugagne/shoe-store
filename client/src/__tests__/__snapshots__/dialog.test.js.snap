// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`dialog.js renders 1`] = `
<WithStyles(Dialog)
  fullWidth={true}
  onClose={[Function]}
  open={true}
  scroll="paper"
>
  <WithStyles(AppBar)
    color="default"
    position="static"
  >
    <WithStyles(Toolbar)>
      <WithStyles(IconButton)
        color="inherit"
      >
        <pure(StoreTwoTone) />
      </WithStyles(IconButton)>
      <WithStyles(Typography)
        color="inherit"
        variant="title"
      >
        Test Title
      </WithStyles(Typography)>
    </WithStyles(Toolbar)>
  </WithStyles(AppBar)>
  <WithStyles(DialogContent)>
    <WithStyles(ShoeRow)
      data={Object {}}
    />
  </WithStyles(DialogContent)>
  <WithStyles(DialogActions)>
    <WithStyles(Button)
      autoFocus={true}
      color="primary"
      onClick={[Function]}
    >
      Close
    </WithStyles(Button)>
  </WithStyles(DialogActions)>
</WithStyles(Dialog)>
`;
